test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common.Linux
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Serilog;
+
+    public class PackageManagement
+    {
+        Lazy<Task<LinuxStandardBase>> lsbTask =
+            new Lazy<Task<LinuxStandardBase>>(
+                async () =>
+                {
+                    var cts = new CancellationTokenSource();
+                    cts.CancelAfter(10000);
+                    string[] platformInfo = await Process.RunAsync("lsb_release", "-sir", cts.Token);
+                    if (platformInfo.Length == 1)
+                    {
+                        platformInfo = platformInfo[0].Split(' ');
+                    }
+
+                    string os = platformInfo[0].Trim();
+                    string version = platformInfo[1].Trim();
+                    SupportedPackageExtension packageExtension = SupportedPackageExtension.None;
+
+                    switch (os)
+                    {
+                        case "Ubuntu":
+                            os = "ubuntu";
+                            packageExtension = SupportedPackageExtension.Deb;
+                            break;
+                        case "Raspbian":
+                            os = "debian";
+                            version = "stretch";
+                            packageExtension = SupportedPackageExtension.Deb;
+                            break;
+                        case "CentOS":
+                            os = os.ToLower();
+                            version = version.Split('.')[0];
+                            packageExtension = SupportedPackageExtension.Rpm;
+
+                            if (version != "7")
+                            {
+                                throw new NotImplementedException($"Don't know how to install daemon on operating system '{os} {version}'");
+                            }
+
+                            break;
+                        default:
+                            throw new NotImplementedException($"Don't know how to install daemon on operating system '{os}'");
+                    }
+
+                    return new LinuxStandardBase(os, version, packageExtension);
+                });
+
+        string forceInstallConfigCmd = null;
+        public string ForceInstallConfigCmd
```

Comment:
Most of these properties are only used internally, so they shouldn't be public.

Reply:
private variable they are!